### ISSUE TITLE
UCI-HAR dataset processing

### ISSUE THREAD

**Issue by XueBaolu** [usc-sail/fed-multimodal#6]
Hey, thank you for your fl_multimodal framework. 
I read the codes for the task of UCI-HAR, and have a question about the process of data. Why does the  MMDatasetGenerator class in dataloader_manager.py discard the y,z  data and keep only x for training? Would this compromise the performance of model?
Looking forward to your reply, thanks again!

**Comment by tiantiaf0627** [comment]
Hello there, I am wondering which line of code you are referring to. If you run the code, you will find that we did not discard y,z data for training, it should be x,y,z data for training.

**Comment by XueBaolu** [comment]
Thanks for your respond. But it seems that the codes on line 100 of the dataloader_manager.py file discards the y and z data. 
The code is,
 ```
data_a = self.modalityA[item][-1]
 if data_a is not None:
   if len(data_a.shape) == 3:
        data_a = data_a[0]
    data_a = torch.tensor(data_a)
    len_a = len(data_a)
```

**Comment by tiantiaf0627** [comment]
I think in your described case, it should be data_a.shape[0] == 3 and data_a = data_a[0, :]. I believe this line of code is for some embedding that has extra dim at axis 0 when extracted.

**Comment by tiantiaf0627** [comment]
Let me know if this answers your question. But feel free to run the code, and you can set a debugger to see if the data includes all the x,y, and z axes.

**Comment by XueBaolu** [comment]
Thanks for your answers. I know the aim of the if statement is to reduce the dimensions of data_a from 3 to 2, and it still contains the features of x,y,z axes, right?

**Comment by tiantiaf0627** [comment]
Yes, that is correct. The dimension for HAR data should be 3xT or Tx3, so this line will not impact data input.

**Comment by XueBaolu** [comment]
Thank you! Best wishes.
